src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs

```diff
@@ -116,7 +116,7 @@ public async Task<ReturnResult> DeleteItemsAsync(IEnumerable<IStorageItemWithPat
 			var deleteFromRecycleBin = source.Select(item => item.Path).Any(path => recycleBinHelpers.IsPathUnderRecycleBin(path));
 			var canBeSentToBin = !deleteFromRecycleBin && await recycleBinHelpers.HasRecycleBin(source.FirstOrDefault()?.Path);
 
-			if (showDialog && ((!permanently && !canBeSentToBin) || UserSettingsService.PreferencesSettingsService.ShowConfirmDeleteDialog)) // Check if the setting to show a confirmation dialog is on
+			if (showDialog && (!permanently && !canBeSentToBin) && UserSettingsService.PreferencesSettingsService.ShowConfirmDeleteDialog) // Check if the setting to show a confirmation dialog is on
```

Comment:
if (showDialog && ((!permanently && !canBeSentToBin && !deleteFromRecycleBin) || UserSettingsService.PreferencesSettingsService.ShowConfirmDeleteDialog)) // Check if the setting to show a confirmation dialog is on

Reply:
We only want to turn off confirmation when deleting from bin, not in all cases.

Reply:
Behavior should be consistent imo. Users turning the option off don't expect the prompt to show.

Reply:
So also when deleting from e.g a network folder (where the file gets permanently deleted without going to bin)?
I'm not 100% sure that Files behaves like explorer in all cases regarding permanent deletion vs sending to bin.

Reply:
I'm not too concerned with matching File Explorer, the main thing is to follow the setting that users pick in Files.

Reply:
As u wish, I just don't want people complaining "I lost my files b/c I thought it was going to bin but Files deleted it" xD

@ferrariofilippo the initial code was the correct one, sorry.

Reply:
The default setting is to prompt users

Reply:
Problem is if the user turns the setting off expecting we have the same behaviour of explorer when deleting files to bin vs permanent. Perhaps we have, I dont' know.

Reply:
We can mointor feedback and see if it's an issue